cmd/cli/kubectl-kyverno/commands/apply/command.go

```diff
@@ -139,6 +142,7 @@ func Command() *cobra.Command {
 	// currently `set` flag supports variable for single policy applied on single resource
 	cmd.Flags().StringVarP(&applyCommandConfig.UserInfoPath, "userinfo", "u", "", "Admission Info including Roles, Cluster Roles and Subjects")
 	cmd.Flags().StringSliceVarP(&applyCommandConfig.Variables, "set", "s", nil, "Variables that are required")
+	cmd.Flags().StringSliceVarP(&applyCommandConfig.VapParams, "validating-admission-policy-params", "", []string{}, "Path to resource files that act as validating admission policy parameters")
```

Comment:
Same comment as above.

Instead of adding a new flag for parameter resources, we can fetch the parameter resource from the resource file passed in the flag `--resource`. I am not sure if this is a good idea. We can discuss it further in the maintainers/community meeting.

Reply:
On some level this is correct.
At the end of the day the parameter is just a resource. But i think doing it this way helps clear the confusion